NitroxServer/GameLogic/NitroxTimer.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NitroxModel.Logger;
+using NitroxModel.Packets;
+using ProtoBufNet;
+
+namespace NitroxServer.GameLogic
+{
+    [Serializable]
+    [ProtoContract]
+    public class NitroxTimer : System.Timers.Timer
+    {
+        [ProtoMember(1)]
+        public string Key;
+        [ProtoMember(2)]
+        public double? TimeLeft
+        {
+            get
+            {
+                return timeLeft;
+            }
+            set { timeLeft = value; }
+        }
+        [ProtoMember(3)]
+        public double InitialInterval;
+        [ProtoMember(4)]
+        public PlayerManager PlayerManager;
+        [ProtoMember(5)]
+        public StoryEventType EventType;
+        [ProtoIgnore]
+        private double? timeLeft;
+        [ProtoIgnore]
+        private DateTime dueTime;
+
+        public NitroxTimer() : base()
```

Comment:
Mandatory field data should be passed in through the constructor OR make it a private constructor with a static `NitroxTimer.From(...)` method that calls the private ctor as Sunrunner said.